packages/@sanity/form-builder/src/inputs/arrays/ArrayOfObjectsInput/item/ArrayItem.tsx

```diff
@@ -66,6 +67,13 @@ export const ArrayItem = memo(function ArrayItem(props: ArrayInputListItemProps)
   const hasFocusWithin = hasFocusWithinPath(props.focusPath, props.value)
   useScrollIntoViewOnFocusWithin(innerElementRef, hasFocusWithin)
 
+  useDidUpdate(hasFocusAtPath(props.focusPath, props.value), (hadFocus, hasFocus) => {
```

Comment:
i think it's better to store the result from hasFocusAtPath in a variable and then use a vanilla effect to call the focus

Reply:
Curious to hear why do you think it would be better :)

I personally feel this captures the intention pretty well. An `useEffect` here would have to keep track of the previous value since we're not merely responding to _change_ but the transition from `!hadFocus` to `hasFocus`.

Merging this for now, but I'm happy to improve this pattern and change it into something better later on.